### ISSUE TITLE
Javadoc Hover: Generic Types are hardly readable in light/default theme

### ISSUE THREAD

**Issue by laeubi** [eclipse-jdt/eclipse.jdt.ui#1442]
![grafik](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/1331477/3d2c56d4-8d87-4801-a4bb-af3536d864e6)

the light green is hardly readable on white background (not sure why the `<?>` is printed differently) especially as a very thin font is used here. If more than one generic is involved it even becomes worse:

![grafik](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/1331477/44dd7f26-03af-45dd-aa51-5182629d1641)
![grafik](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/1331477/1e9ee7c7-0997-4650-b661-8625c0ff6a6b)

```
Eclipse SDK

Version: 2024-06 (4.32)
Build id: I20240529-1800
```

**Comment by jukzi** [comment]
@RedeemerSK is that something you may work on?

**Comment by iloveeclipse** [comment]
May be using bold font style will be the solution here?

**Comment by laeubi** [comment]
It looks like the last example the `List<R>` ... is using bold font I still find it hard to read on light background.... the colors just seem not suitable for an almost white background.

**Comment by iloveeclipse** [comment]
But it is much better compared to the non-bold R:
![image](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/964108/342764b8-5989-4088-ba55-90f82f51e410)
so using bold style will be an easy fix (I assume), while selecting "right" colors will most likely lead to never ending discussion, as everyone will have other preferences...

**Comment by laeubi** [comment]
I would be even happy with (maybe a preference) to disable the colors ;-)

**Comment by humphreygao** [comment]
It format the method parameter(one parameter per line) to improve readability,  so I like it.

**Comment by laeubi** [comment]
> It format the method parameter(one parameter per line) to improve readability, so I like it.

The problem is **not** the method parameters!

**Comment by iloveeclipse** [comment]
> I would be even happy with (maybe a preference) to disable the colors like it was before ;-)

Have you noticed that: 
![image](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/964108/4e504c78-857b-455a-9b52-43e7973e44ee)

**Comment by laeubi** [comment]
Thanks thats better now, but one need to focus (F2) the hover and at least under Linux its hard to oben the submenu as the hover closes in between and the Color preferences itself are empty

https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/1331477/782bd4d7-64b9-4670-8615-2df8c5aa3363

**Comment by RedeemerSK** [comment]
> Thanks thats better now, but one need to focus (F2) the hover and at least under Linux its hard to oben the submenu as the hover closes in between and the Color preferences itself are empty
> Peek.2024-06-07.14-11.mp4

I have very limited expertise in Linux (distros) native graphics backends and thus I can only wonder why the hover viewer won't stay displayed after opening javadoc styling enhancements menu o.O.

Btw, same menu (button) is available in Javadoc view.




> But it is much better compared to the non-bold R:
![image](https://private-user-images.githubusercontent.com/964108/337613634-342764b8-5989-4088-ba55-90f82f51e410.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3ODIzNjEsIm5iZiI6MTcxNzc4MjA2MSwicGF0aCI6Ii85NjQxMDgvMzM3NjEzNjM0LTM0Mjc2NGI4LTU5ODktNDA4OC1iYTU1LTkwZjgyZjUxZTQxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwN1QxNzQxMDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ODllZmJkMjEyNGYzOWY3OWI0YzZkMjc4NzBiOTRmMjg0YjI4ODZhZDJlYjBjNjRjZWM0NTJmM2E1ZTljZDdjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.2F0dDuWEBUSFD93LeAkJWlnWrJivwaTDuuzsRk6mApE)
so using bold style will be an easy fix (I assume), while selecting "right" colors will most likely lead to never ending discussion, as everyone will have other preferences...

Right now none of the type params in the content is bold. It's just that method return type section is italic which makes it to look little bit bolder.

Shortly I'll post screens how it would look with type params being same font-weight like the rest of the signature - because that's the actual difference right now ... they were made lighter than the rest with the intention to decrease their contribution to visual complexity.

**Comment by RedeemerSK** [comment]
Current:
![javadoc_styling_type_params_less_font-weight_win png](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/36334098/19059ed1-b660-4244-a695-a8441bd3663e)
![javadoc_styling_type_params_less_font-weight_ubuntu](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/36334098/6f795701-e030-400d-851d-87cb7be7316e)

New (same font-weight as rest):
![javadoc_styling_type_params_no_font-weight_win png](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/36334098/36089862-f98e-4834-a8f7-422c753a056a)
![javadoc_styling_type_params_no_font-weight_ubuntu](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/36334098/27e11917-8351-4a06-9592-02bb19900233)

**Comment by laeubi** [comment]
@RedeemerSK with "normal" font weight it is better readable indeed I think one should either use colors or the lighter font weight but not both, and yes the hover/menu seems very "buggy"...

**Comment by RedeemerSK** [comment]
@laeubi please what's your platform ? (linux distro / desktop environment / anything that may have something to do with how SWT renders menus). I assume you are running it natively, not in VM or some other non-native way, right ?

Also ... until the fix that removes lighter text makes it into future builds, for those that are impatient and also willing to walk the unlawful path and do little bit of hacking that I of course won't officially suggest (\*wink\* \*wink\*), there's a way to apply the fix to any existing eclipse installation.

[The fix](https://github.com/RedeemerSK/eclipse.jdt.ui/commit/4c434a981eb4fd224081bfa2833f7d3bfe80ebd6) can be manually applied to the files bundled inside org.eclipse.jdt.ui plugin JAR used by eclipse installation (present in ~/.p2/pool/plugins directory). Surprisingly for me, because since the JAR is signed, I (it seems wrongly) assumed tampering with it would render it invalid / unusable, which is not the case and eclipse has no problem starting with modified plugin JAR.

**Comment by RedeemerSK** [comment]
> Thanks thats better now, but one need to focus (F2) the hover and at least under Linux its hard to oben the submenu as the hover closes in between and the Color preferences itself are empty

Just noticed that hover & menu behaves differently when activated by pressing F2 (then going to menu hides the hover) compared to being activated by "entering into it" with mouse (then going to menu keeps hover displayed) on Ubuntu I'm testing this on. Will look into that. Can others confirm same behavior, @laeubi  ?

**Comment by RedeemerSK** [comment]
Fix PR: #1482. I hope it's OK I am also fixing bugged styling enhancements menu in that PR and not just the hard to read colored type parameters. If necessary, I can split into two separate PRs.
